addons/mail/models/mail_channel.py

```diff
@@ -929,20 +948,27 @@ def channel_fetch_preview(self):
         """ Return the last message of the given channels """
         if not self:
             return []
-        self._cr.execute("""
-            SELECT mail_channel_id AS id, MAX(mail_message_id) AS message_id
-            FROM mail_message_mail_channel_rel
-            WHERE mail_channel_id IN %s
-            GROUP BY mail_channel_id
-            """, (tuple(self.ids),))
-        channels_preview = dict((r['message_id'], r) for r in self._cr.dictfetchall())
+        channels_last_message_ids = self._channel_last_message_ids()
+        channels_preview = dict((r['message_id'], r) for r in channels_last_message_ids)
         last_messages = self.env['mail.message'].browse(channels_preview).message_format()
         for message in last_messages:
             channel = channels_preview[message['id']]
             del(channel['message_id'])
             channel['last_message'] = message
         return list(channels_preview.values())
 
+    def _channel_last_message_ids(self):
+        """ Return the last message of the given channels."""
+        if not self:
+            return []
+        self.env.cr.execute("""
```

Comment:
Even if you didn't have the issue with it, executing a query without calling `flush` before is really weak.
Currently there might no data necessary in the query that is updated in cache previously to the request, but there is no guarantee it will always be the case.

Reply:
```
self.flush()
```
?

Reply:
Yes